nixos/modules/programs/firefox.nix

```diff
@@ -307,44 +331,34 @@ in
       );
       programs.firefox.nativeMessagingHosts.packages = forEachEnabledNmh (_: v: v.package);
 
-      environment.systemPackages = [
-        (cfg.package.override (old: {
-          extraPrefsFiles =
-            old.extraPrefsFiles or [ ]
-            ++ cfg.autoConfigFiles
-            ++ [ (pkgs.writeText "firefox-autoconfig.js" cfg.autoConfig) ];
-          nativeMessagingHosts = old.nativeMessagingHosts or [ ] ++ cfg.nativeMessagingHosts.packages;
-          cfg = (old.cfg or { }) // cfg.wrapperConfig;
-        }))
-      ];
+      environment.systemPackages = [ cfg.finalPackage ];
 
       environment.etc =
         let
           policiesJSON = policyFormat.generate "firefox-policies.json" { inherit (cfg) policies; };
         in
         lib.mkIf (cfg.policies != { }) {
-          "firefox/policies/policies.json".source = "${policiesJSON}";
-        };
+          "firefox/policies/policies.json".source = policiesJSON;
 
-      # Preferences are converted into a policy
-      programs.firefox.policies = {
-        DisableAppUpdate = true;
-        Preferences = (
-          builtins.mapAttrs (_: value: {
-            Value = value;
-            Status = cfg.preferencesStatus;
-          }) cfg.preferences
-        );
-        ExtensionSettings = builtins.listToAttrs (
-          builtins.map (
-            lang:
-            lib.attrsets.nameValuePair "langpack-${lang}@firefox.mozilla.org" {
-              installation_mode = "normal_installed";
-              install_url = "https://releases.mozilla.org/pub/firefox/releases/${cfg.package.version}/linux-x86_64/xpi/${lang}.xpi";
-            }
-          ) cfg.languagePacks
-        );
-      };
+          # Preferences are converted into a policy
+          programs.firefox.policies = {
+            DisableAppUpdate = true;
+            Preferences = builtins.mapAttrs (_: value: {
+              Value = value;
+              Status = cfg.preferencesStatus;
+            }) cfg.preferences;
+            ExtensionSettings = builtins.listToAttrs (
+              builtins.map (
+                lang:
+                lib.nameValuePair "langpack-${lang}@firefox.mozilla.org" {
+                  installation_mode = "normal_installed";
+                  install_url = "https://releases.mozilla.org/pub/firefox/releases/${cfg.package.version}/linux-x86_64/xpi/${lang}.xpi";
+                }
+              ) cfg.languagePacks
+            );
+          };
+
+        };
```

Comment:
I am not sure this (and the other) reformatting should be in the same PR (and commit!) as the added feature.

That said, the changes look fine to me.